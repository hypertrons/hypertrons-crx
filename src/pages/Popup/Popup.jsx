import React from 'react';
import logo from '../../assets/img/logo.svg';
import Greetings from '../../components/Greetings/Greetings';
import './Popup.css';

const Popup = () => {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/pages/Popup/Popup.jsx</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React!
        </a>
      </header>
    </div>
  );
};

export default Popup;
