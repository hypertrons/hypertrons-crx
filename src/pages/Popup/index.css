body {
  min-width: 200px;
}