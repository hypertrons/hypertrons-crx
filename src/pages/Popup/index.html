<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title></title>
  </head>

  <body>
    <div id="app-container">
      Hello
    </div>
  </body>
</html>
