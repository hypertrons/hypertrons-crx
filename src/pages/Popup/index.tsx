import React from 'react';
import { render } from 'react-dom';
import PopupPage from '../../components/PopupPage/index'

import './index.css';

render(<PopupPage />, window.document.querySelector('#app-container'));
