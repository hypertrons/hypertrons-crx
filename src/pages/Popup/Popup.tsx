import React, { useEffect, useState } from 'react';
import {
  DefaultButton,
  Image,
  ImageFit,
  Stack,
  Text,
  Toggle,
} from 'office-ui-fabric-react';
import './Popup.css';
import Settings, { loadSettings } from '../../utils/settings';
import MetaData, { loadMetaData } from '../../utils/metadata';
import { chromeSet, getMessageByLocale } from '../../utils/utils';

const Popup: React.FC = () => {
  const [settings, setSettings] = useState(new Settings());
  const [metaData, setMetaData] = useState(new MetaData());
  const [inited, setInited] = useState(false);

  useEffect(() => {
    const initSettings = async () => {
      const temp = await loadSettings();
      setSettings(temp);
      setInited(true);
    };
    if (!inited) {
      initSettings();
    }
  }, [inited, settings]);

  useEffect(() => {
    const initMetaData = async () => {
      const temp = await loadMetaData();
      setMetaData(temp);
    };
    initMetaData();
  }, []);

  const saveSettings = async (settings: Settings) => {
    setSettings(settings);
    await chromeSet('settings', settings.toJson());
  };

  if (!inited) {
    return <div />;
  }

  return (
    <Stack horizontalAlign="center">
      <Stack
        horizontalAlign="space-around"
        verticalAlign="center"
        style={{ margin: '5px', padding: '3px' }}
        tokens={{
          childrenGap: 10,
        }}
      >
        <Stack horizontalAlign="center">
          <Toggle
            label={getMessageByLocale(
              'options_enable_toggle_autoCheck',
              settings.locale
            )}
            defaultChecked={settings.isEnabled}
            onText={getMessageByLocale('global_toggle_onText', settings.locale)}
            offText={getMessageByLocale(
              'global_toggle_offText',
              settings.locale
            )}
            onChange={async (e, checked) => {
              settings.isEnabled = checked;
              await saveSettings(settings);
            }}
          />
        </Stack>
        {metaData.token !== '' && (
          <Stack
            horizontal
            verticalAlign="center"
            style={{
              margin: '5px',
              padding: '3px',
              width: '200px',
            }}
            tokens={{
              childrenGap: 5,
            }}
          >
            <Image
              width={75}
              height={75}
              src={metaData.avatar}
              imageFit={ImageFit.centerCover}
            />
            <Text
              variant="large"
              style={{ marginLeft: 25, width: 100, wordWrap: 'break-word' }}
            >
              {metaData.name}
            </Text>
          </Stack>
        )}
        {metaData.token === '' && (
          <DefaultButton
            onClick={() => {
              chrome.runtime.openOptionsPage();
            }}
            style={{
              width: 120,
            }}
          >
            {getMessageByLocale('options_token_title', settings.locale)}
          </DefaultButton>
        )}
      </Stack>
    </Stack>
  );
};

export default Popup;
