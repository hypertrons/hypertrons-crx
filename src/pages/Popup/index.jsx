import React from 'react';
import { render } from 'react-dom';

import Popup from './Popup';
import './index.css';

render(<Popup />, window.document.querySelector('#app-container'));

if (module.hot) module.hot.accept();
