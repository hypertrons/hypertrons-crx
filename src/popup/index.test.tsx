import React from 'react';
import renderer from 'react-test-renderer';
import Popup from './index';

it('renders correctly', () => {
  const tree = renderer.create(<Popup />).toJSON();
  expect(tree).toMatchSnapshot();
});
