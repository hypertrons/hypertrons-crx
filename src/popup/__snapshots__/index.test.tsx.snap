// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Basic configuration
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal"
        id="basic"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="Welcome Message"
            >
              Welcome Message
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <button
                  aria-checked={true}
                  className="ant-switch ant-switch-checked"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="switch"
                  type="button"
                >
                  <div
                    className="ant-switch-handle"
                  />
                  <span
                    className="ant-switch-inner"
                  >
                    on
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-label"
            style={Object {}}
          >
            <label
              className=""
              title="Showing Dashboard"
            >
              Showing Dashboard
            </label>
          </div>
          <div
            className="ant-col ant-form-item-control"
            style={Object {}}
          >
            <div
              className="ant-form-item-control-input"
            >
              <div
                className="ant-form-item-control-input-content"
              >
                <button
                  aria-checked={true}
                  className="ant-switch ant-switch-checked"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="switch"
                  type="button"
                >
                  <div
                    className="ant-switch-handle"
                  />
                  <span
                    className="ant-switch-inner"
                  >
                    on
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Identity Authentication and Management
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-tabs ant-tabs-top"
      >
        <div
          className="ant-tabs-nav"
          onKeyDown={[Function]}
          role="tablist"
        >
          <div
            className="ant-tabs-nav-wrap"
          >
            <div
              className="ant-tabs-nav-list"
              style={
                Object {
                  "transform": "translate(0px, 0px)",
                  "transition": undefined,
                }
              }
            >
              <div
                className="ant-tabs-tab ant-tabs-tab-active"
                onClick={[Function]}
                style={
                  Object {
                    "marginRight": undefined,
                  }
                }
              >
                <div
                  aria-controls={null}
                  aria-selected={true}
                  className="ant-tabs-tab-btn"
                  id={null}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="tab"
                  tabIndex={0}
                >
                  <span>
                    <span
                      aria-label="github"
                      className="anticon anticon-github"
                      role="img"
                      style={
                        Object {
                          "fontSize": "20px",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="github"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                        />
                      </svg>
                    </span>
                    <span
                      style={
                        Object {
                          "fontSize": "18px",
                        }
                      }
                    >
                      GitHub
                    </span>
                  </span>
                </div>
              </div>
              <div
                className="ant-tabs-tab"
                onClick={[Function]}
                style={
                  Object {
                    "marginRight": undefined,
                  }
                }
              >
                <div
                  aria-controls={null}
                  aria-selected={false}
                  className="ant-tabs-tab-btn"
                  id={null}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="tab"
                  tabIndex={0}
                >
                  <span>
                    <span
                      aria-label="gitlab"
                      className="anticon anticon-gitlab"
                      role="img"
                      style={
                        Object {
                          "fontSize": "20px",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="gitlab"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M913.9 552.2L805 181.4v-.1c-7.6-22.9-25.7-36.5-48.3-36.5-23.4 0-42.5 13.5-49.7 35.2l-71.4 213H388.8l-71.4-213c-7.2-21.7-26.3-35.2-49.7-35.2-23.1 0-42.5 14.8-48.4 36.6L110.5 552.2c-4.4 14.7 1.2 31.4 13.5 40.7l368.5 276.4c2.6 3.6 6.2 6.3 10.4 7.8l8.6 6.4 8.5-6.4c4.9-1.7 9-4.7 11.9-8.9l368.4-275.4c12.4-9.2 18-25.9 13.6-40.6zM751.7 193.4c1-1.8 2.9-1.9 3.5-1.9 1.1 0 2.5.3 3.4 3L818 394.3H684.5l67.2-200.9zm-487.4 1c.9-2.6 2.3-2.9 3.4-2.9 2.7 0 2.9.1 3.4 1.7l67.3 201.2H206.5l57.8-200zM158.8 558.7l28.2-97.3 202.4 270.2-230.6-172.9zm73.9-116.4h122.1l90.8 284.3-212.9-284.3zM512.9 776L405.7 442.3H620L512.9 776zm157.9-333.7h119.5L580 723.1l90.8-280.8zm-40.7 293.9l207.3-276.7 29.5 99.2-236.8 177.5z"
                        />
                      </svg>
                    </span>
                     
                    <span
                      style={
                        Object {
                          "fontSize": "18px",
                        }
                      }
                    >
                      GitLab
                    </span>
                     
                  </span>
                </div>
              </div>
              <div
                className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              />
            </div>
          </div>
          <div
            className="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
          >
            <button
              aria-controls="null-more-popup"
              aria-expanded={false}
              aria-haspopup="listbox"
              aria-hidden="true"
              className="ant-tabs-nav-more"
              id="null-more"
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "marginRight": undefined,
                  "order": 1,
                  "visibility": "hidden",
                }
              }
              tabIndex={-1}
              type="button"
            >
              <span
                aria-label="ellipsis"
                className="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          className="ant-tabs-content-holder"
        >
          <div
            className="ant-tabs-content ant-tabs-content-top"
            style={null}
          >
            <div
              aria-hidden={false}
              aria-labelledby={null}
              className="ant-tabs-tabpane ant-tabs-tabpane-active"
              id={null}
              role="tabpanel"
              style={Object {}}
              tabIndex={0}
            >
              <div>
                <div>
                  <p>
                    Great！You have connected to
                     
                    <span
                      style={
                        Object {
                          "color": "deepskyblue",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      GitHub
                    </span>
                    ! Now you can add or update your personal access token below：
                  </p>
                  <div
                    style={
                      Object {
                        "marginTop": "20px",
                      }
                    }
                  >
                    <form
                      className="ant-form ant-form-horizontal"
                      id="basic"
                      onSubmit={[Function]}
                    >
                      <div
                        className="ant-row ant-form-item"
                        style={Object {}}
                      >
                        <div
                          className="ant-col ant-form-item-label"
                          style={Object {}}
                        >
                          <label
                            className=""
                            htmlFor="basic_token"
                            title="Token"
                          >
                            Token
                          </label>
                        </div>
                        <div
                          className="ant-col ant-form-item-control"
                          style={Object {}}
                        >
                          <div
                            className="ant-form-item-control-input"
                          >
                            <div
                              className="ant-form-item-control-input-content"
                            >
                              <input
                                className="ant-input"
                                id="basic_token"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className="ant-row ant-form-item"
                        style={Object {}}
                      >
                        <div
                          className="ant-col ant-col-14 ant-col-offset-2 ant-form-item-control"
                          style={Object {}}
                        >
                          <div
                            className="ant-form-item-control-input"
                          >
                            <div
                              className="ant-form-item-control-input-content"
                            >
                              <button
                                className="ant-btn ant-btn-primary"
                                onClick={[Function]}
                                type="submit"
                              >
                                <span>
                                  Save
                                </span>
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              aria-labelledby={null}
              className="ant-tabs-tabpane"
              id={null}
              role="tabpanel"
              style={
                Object {
                  "display": "none",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
